src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 0;
+    private static final int DEFAULT_SIZE = 0;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+        size = DEFAULT_SIZE;
+    }
+
+    private void ensureCapacity(int minCapacity) {
+        if (minCapacity > array.length) {
+            int newCapacity = array.length + (array.length >> 1);
+            if (newCapacity < minCapacity) {
+                newCapacity = minCapacity;
+            }
```

Comment:
I think this code would be redundant if size of initial array would be 10 for example.

Reply:
Yes, I can agree with you, but for me it was necessary to understand the logic of this method. It is crucial for dynamically resizing the array when adding elements, reducing unnecessary resizing operations, and the growth factor that impacts performance and memory usage.